src/registry/map/edit.tsx

```diff
@@ -52,6 +53,24 @@ const EditForm: EditFormDefinition<MapComponentSchema> = () => {
     }
   });
 
+  // Make sure each overlay has a unique internal id.
+  useEffect(() => {
+    if (!values.overlays || !values.overlays.length) return;
+
+    const overlays: ExtendedMapOverlay[] = values.overlays;
+    // Only update if at least one overlay is missing an internal id
+    const needsUpdate = overlays.some(o => !o._OF_INTERNAL_id);
+    if (!needsUpdate) return;
+
+    setValues({
+      ...values,
+      overlays: overlays.map(o => ({
+        ...o,
+        _OF_INTERNAL_id: o._OF_INTERNAL_id ?? crypto.randomUUID(),
+      })),
+    });
```

Comment:
can't you `setValue` targetting just the `overlays` field? the object spread of `...values` breaks all references and causes a re-render of *every* field now, which is quite inefficient.

I also think that calling this effect only once is flawed, if someone pastes an old configuration in the JSON panel, this effect won't fire again and it re-introduces the problem. Perhaps this should be moved down into the component that renders a single layer panel and make it self-correct based on whether a unique ID is present or not.